### PR TITLE
fix(ui): disambiguate sidebar entries when app keys share a display name

### LINKED ISSUE #1452: Disambiguate sidebar entries when app keys share a class name
## Description

When two separate app keys use the same class (e.g. `blocking_io_lab` and `blocking_io_lab_ignore` both configured with `class_name = "BlockingIOLab"`, neither setting an explicit `display_name`), the sidebar shows two entries with identical labels and no way to tell them apart. `AppManifest.display_name` falls back to `class_name` when unset (`config/classes.py:168`), so both manifests end up with the same `display_name`, and the sidebar renders `manifest.display_name` verbatim with no uniqueness check.

This is distinct from the existing `[[config]]` multi-instance pattern (e.g. a `RemoteApp` with several instances), which the UI already handles correctly by grouping instances under one entry with an expandable instance selector — that case is one app key with many instances of the same class. The bug case here is two *different* app keys, each with its own manifest, config, and lifecycle, that happen to share an implementation class. That's a legitimate use case (same code, conceptually different apps with different config) and the sidebar should degrade gracefully instead of showing indistinguishable rows.

Expected: when `display_name` collides across two or more entries in the app list, the sidebar should fall back to a disambiguating label (app key, or instance_name where applicable) instead of showing identical, unclickable-apart entries.

## Decision (2026-09-14)

Two implementation calls, decided:

1. **Replace the label with `app_key`, don't append it.** `AppEntry`'s label renders in a single `truncate`-clamped `<span>` in a narrow (collapsible) sidebar (`sidebar.tsx:73`). Appending `" (app_key)"` to a long `display_name` risks the truncation ellipsis eating exactly the disambiguating suffix — defeating the fix in the case it exists to handle. `app_key` is already guaranteed unique, so replacing is simpler and truncation-safe.
2. **Detect collisions against the full manifest list, not the search-filtered one.** `groupAndSortApps` (`sidebar-groups.ts:34`) is currently called with `filtered` (post-search) at the call site (`sidebar.tsx:202`). Computing collisions against that would make a colliding app's label flip between `display_name` and `app_key` depending on whether the *other* colliding entry still matches the current search text — a confusing flicker unrelated to the actual bug. Compute the duplicate set once against `manifestsData` (`sidebar.tsx:185`, the full list before search filtering), independent of search state.

## Fix Shape

**`sidebar-groups.ts`** — new exported helper:

```ts
/** Display names that appear on 2+ manifests -- these need a disambiguating label instead
 * of the bare (colliding) display_name. Computed against the full manifest list so a given
 * app's label doesn't flip based on unrelated search-filtering state. */
export function findDuplicateDisplayNames(manifests: AppManifest[]): Set<string> {
  const counts = new Map<string, number>();
  for (const m of manifests) {
    counts.set(m.display_name, (counts.get(m.display_name) ?? 0) + 1);
  }
  return new Set([...counts.entries()].filter(([, count]) => count > 1).map(([name]) => name));
}
```

**`sidebar.tsx`**:

```ts
// Computed once against the full list (manifestsData), not `filtered` -- see #1452.
const duplicateDisplayNames = findDuplicateDisplayNames(manifestsData);
```

```tsx
<AppEntry
  manifest={manifest}
  location={location}
  searchString={searchString}
  appStatuses={appStatus}
  labelIsAmbiguous={duplicateDisplayNames.has(manifest.display_name)}
/>
```

`AppEntry` (`sidebar.tsx:36-73`):

```ts
interface AppEntryProps {
  manifest: AppManifest;
  location: string;
  searchString: string;
  appStatuses: Record<string, AppStatusEntry>;
  labelIsAmbiguous: boolean;
}

function AppEntry({ manifest, location, searchString, appStatuses, labelIsAmbiguous }: AppEntryProps) {
  ...
  const label = labelIsAmbiguous ? manifest.app_key : manifest.display_name;
  ...
  <span className="min-w-0 flex-1 truncate">{label}</span>
  ...
  aria-label={expanded ? `Collapse ${label}` : `Expand ${label}`}
```

(Update the `aria-label` expand/collapse strings too, at `sidebar.tsx:85`, so screen-reader text matches the disambiguated label rather than reverting to the ambiguous `display_name`.)

The multi-instance `instanceList` rendering (`sidebar.tsx:93-124`) is untouched — it already disambiguates via `inst.instance_name`, a separate mechanism for a separate case (one app key, many instances), and this fix must not be confused with or interfere with it, per the issue's own framing.

## Key Items

- `frontend/src/components/layout/sidebar-groups.ts` — add `findDuplicateDisplayNames()`.
- `frontend/src/components/layout/sidebar.tsx:36-92,185,202,324` — compute the duplicate set against `manifestsData`, thread `labelIsAmbiguous` prop into `AppEntry`, swap the label and `aria-label` strings.
- `frontend/src/components/layout/sidebar.test.tsx` — new regression test for two distinct app keys with a colliding `display_name`.

## Acceptance Criteria

- [ ] When two or more manifests in the sidebar's app list share the same `display_name`, each affected entry displays `app_key` instead of the bare, duplicated `display_name`.
- [ ] Entries with a unique `display_name` are unaffected — no visual change for the common case.
- [ ] The existing `[[config]]` multi-instance grouping (single app key, `instance_count > 1`, expandable instance list) continues to work unchanged.
- [ ] A colliding entry's label does not change when the sidebar search text changes (collision detection is search-independent).
- [ ] The expand/collapse `aria-label` on a colliding multi-instance entry uses the disambiguated label, not the raw `display_name`.
- [ ] New test in `sidebar.test.tsx` covering two distinct app keys with a colliding `display_name`, asserting both entries render distinguishable text.
- [ ] Existing sidebar tests continue to pass.

## Affected Areas

- `frontend/src/components/layout/sidebar.tsx` — `AppEntry` renders `manifest.display_name` directly with no collision check.
- `frontend/src/components/layout/sidebar-groups.ts` — `groupAndSortApps` sorts entries by `display_name` alone; `findDuplicateDisplayNames()` added here per the decision above.
- `frontend/src/components/layout/sidebar.test.tsx` — add a regression test for the collision case.
- `src/hassette/config/classes.py:168` — root cause of the collision: `AppManifest.display_name` defaults to `class_name` when the user doesn't set an explicit `display_name`, so two app keys sharing a class silently produce identical labels. Not changed by this fix — the frontend degrades gracefully instead.

## Context

- The multi-instance case (`manifest.instance_count > 1`) is a single app key with multiple `[[config]]` blocks of the same class; the sidebar already disambiguates those via `instance_name` in the expandable instance list. Do not conflate that path with this one.
- The two-app-keys-one-class scenario has no existing regression coverage in `sidebar.test.tsx` — all existing fixtures use distinct `display_name` values per app key.
- Fix belongs on the frontend: the backend manifest validator (`config/classes.py`) only sees one app's config at a time and has no visibility into whether its `class_name`-derived `display_name` collides with another app key's manifest. The sidebar already receives the full app list (`useManifests()` → `allManifests`), so it's the natural place to detect and resolve the collision.

### PR DESCRIPTION
**Pull request by NodeJSmith**
## Summary

When two or more app keys share the same `display_name` (e.g. both
configured with the same class and no explicit `display_name`), the sidebar
showed identical labels for different apps, making them indistinguishable.

This fix detects display-name collisions across the full manifest list and
falls back to showing `app_key` instead of `display_name` for any entry
whose label would otherwise be ambiguous. Collision detection runs against the
unfiltered manifest list so a given app's label doesn't flip as the user types
into the search box.

## Changes

- **`sidebar-groups.ts`**: new `findDuplicateDisplayNames()` — scans all
  manifests and returns the set of display names that appear more than once.
- **`sidebar.tsx`**: `Sidebar` computes the duplicate set once from
  `liveManifests`, passes `labelIsAmbiguous` to each `AppEntry`, which
  selects `app_key` or `display_name` accordingly. Aria labels updated to
  match the visible label.
- **`sidebar-groups.test.ts`**: unit tests for `findDuplicateDisplayNames`
  (unique names, collisions, edge cases).
- **`sidebar.test.tsx`**: component tests verifying disambiguation renders
  `app_key` for colliding entries, leaves unique entries alone, and that
  collision detection is independent of search filtering.

## Testing

Targeted frontend tests run locally (57 tests, all passing):
- `sidebar-groups.test.ts` — unit tests for the new helper
- `sidebar.test.tsx` — component-level rendering tests for the collision behavior

CI runs the full backend + frontend + e2e suite.

## Screenshots

Sidebar with disambiguated app keys

Closes #1452

### CODE REVIEW

**Comment by NodeJSmith** [comment]
@coderabbitai review

File: frontend/src/components/layout/sidebar.tsx [hunk 1]
```diff
@@ -70,7 +72,7 @@ function AppEntry({ manifest, location, searchString, appStatuses }: AppEntryPro
             data-testid="app-link"
           >
             <StatusShape kind={kind} size={STATUS_SHAPE_SIZE} />
-            <span className="min-w-0 flex-1 truncate">{manifest.display_name}</span>
+            <span className="min-w-0 flex-1 truncate">{label}</span>
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply frontend/src/components/layout/sidebar.tsx:75]: Skipping — the PR already improves the pre-existing state (where both entries showed the *identical* display_name with zero disambiguation). With app_key shown instead, the entries are always unique. End-truncation could theoretically hide a distinguishing suffix for very long same-prefix keys, but hassette app keys are typically short enough that this is a negligible edge case. Middle truncation would add CSS complexity without a real-world payoff.
